car.go

```diff
@@ -159,68 +174,104 @@ func (cr *CarReader) Next() (blocks.Block, error) {
 		return nil, fmt.Errorf("mismatch in content integrity, name: %s, data: %s", c, hashed)
 	}
 
-	return blocks.NewBlockWithCid(data, c)
+	offset := cr.offset + l - uint64(len(data))
+	cr.offset += l
+
+	blk, err := blocks.NewBlockWithCid(data, c)
+	if cr.fullPath == "" {
+		return blk, err
+	}
+	nd, err := format.Decode(blk)
+	if err != nil {
+		return nil, err
+	}
+	return &posinfo.FilestoreNode{
+		Node: nd,
+		PosInfo: &posinfo.PosInfo{
+			Offset:   offset,
+			FullPath: cr.fullPath,
+			Stat:     cr.stat,
+		},
+	}, nil
 }
 
 type batchStore interface {
 	PutMany([]blocks.Block) error
 }
 
 func LoadCar(s Store, r io.Reader) (*CarHeader, error) {
+	logger.Debug("will load car now")
 	cr, err := NewCarReader(r)
 	if err != nil {
 		return nil, err
 	}
 
 	if bs, ok := s.(batchStore); ok {
+		logger.Debug("will load car fast")
 		return loadCarFast(bs, cr)
 	}
 
+	logger.Debug("will load car slow")
 	return loadCarSlow(s, cr)
 }
 
 func loadCarFast(s batchStore, cr *CarReader) (*CarHeader, error) {
+	nBlocks := 0
 	var buf []blocks.Block
 	for {
 		blk, err := cr.Next()
 		switch err {
 		case io.EOF:
 			if len(buf) > 0 {
 				if err := s.PutMany(buf); err != nil {
+					logger.Errorf("failed to write %d blocks on completion, nBlocksWritten=%d,  err=%s", len(buf), nBlocks, err)
```

Comment:
Since you are using `go-log` you may want to consider structured logging with `logger.Errorw()`, like this:
```go
logger.Errorw("failed to write blocks on completion", "totalBlocks", len(buf), "nBlocksWritten", nBlocks, "err", err)
```